src/main/java/br/edu/ufcg/computacao/alumni/core/Match.java

```diff
@@ -0,0 +1,238 @@
+			return Curso.DOUTORADO_EM_CIENCIA_DA_COMPUTACAO;
+		}
+		if (fieldInformaticaSet.contains(field) && degreeMestradoSet.contains(degree)) {
+			return Curso.MESTRADO_EM_INFOMATICA;
+		}
+		if (fieldProcDadosSet.contains(field) && degreeGraduacaoSet.contains(degree)) {
+			return Curso.GRADUACAO_PROCESSAMENTO_DE_DADOS;
+		}
+		return null;
+	}
+	
+	private String[] filterName(String name) {
+		Set<String> specialNames = new HashSet<>(Arrays.asList(new String[] {"DE", "DA", "O", "A", "E"}));
+		String[] splitedName = name.split(" ");
+		String[] splitedFilteredName = new String[splitedName.length];
+		
+		for (int i = 0; i < splitedName.length; i++) {
+			String nome = splitedName[i].toUpperCase();
+			if (!specialNames.contains(nome)) { 
+				splitedFilteredName[i] = nome;
+			}
+		}
+		return splitedFilteredName;
+	}
+	
+	private int getNamesLength(String[] splitedName) {
+		int cont = 0;
+		for (String nome : splitedName) {
+			cont += nome == null ? 0 : 1;
+		}
+		return cont;
+	}
+	
+	private String getFirstName(String name) {
+		Set<String> specialNames = new HashSet<>(Arrays.asList(new String[] {"JUNIOR", "JR", "NETO", "SOBRINHO"}));
+		String[] splitedName = filterName(name);
+		int cont = 0;
+		
+		for (String nome : splitedName) { 
+			if (specialNames.contains(nome)) {
+				cont++;
+			}
+		}
+		
+		if (getNamesLength(splitedName) - cont <= 4) {
+			return (splitedName[0] + " " + splitedName[1]).trim();
+		} else {
+			return splitedName[0].trim();
+		}
+	}
+	
+	private String getLastName(String name) {
+		List<String> names = Arrays.asList(filterName(name));
+		List<String> firstNameList = Arrays.asList(getFirstName(name).split(" "));
+
+		List<String> lastNameList = names.stream().filter(n -> !firstNameList.contains(n)).collect(Collectors.toList());
+		String lastName = lastNameList.stream().reduce("", (a, b) -> a + " " + b).trim();
+		
+		return lastName;
+	}
+
+	private int getScoreFromName(String alumniName, String linkedinName) {
+		if (alumniName.equals(linkedinName)) return 100;
+		
+		String alumniFirstName = getFirstName(alumniName);
+		String alumniLastName = getLastName(alumniName);
+		
+		String linkedinFirstName = getFirstName(linkedinName);
+		String linkedinLastName = getLastName(linkedinName);
+		
+		int score = 0;
+		if (alumniFirstName.equals(linkedinFirstName)) score += 60;
+		if (alumniLastName.equals(linkedinLastName)) score += 40;
+		return score;
+	}
+
+	private int getScoreFromYear(String alumniYear, String linkedinYear) {
+		if (linkedinYear == null) return 0;
+		if (alumniYear.contains(linkedinYear)) return 1;
+		return 0;
+	}
+	
+	private Grau getGrauData(Grau[] alumniGraus, Curso curso) {
+		return Arrays.asList(alumniGraus)
+				.stream()
+				.filter(grau -> grau.getCurso().equals(curso))
+				.findAny()
+				.orElse(null);
+	}
+	
+	private int getMatchesFromCurso(UfcgAlumnusData alumni, LinkedinSchoolData schoolData, String schoolUrl) {
```

Comment:
A assinatura dessa função deveria ser diferente:
private int getMatchesFromCurso(UfcgAlumnusData alumni, LinkedinSchoolData schoolData, SchoolName school).

A classe SchoolName tem um vetor de nomes e um vetor de DateRange que indica a validade de cada nome. No caso da UFCG seria:
A url da UFPB (https://www.linkedin.com/school/ufpb/), DateRange (Jan 1950, Mar 2002), isCurrent (false)
A url da UFCG (https://www.linkedin.com/school/universidade-federal-de-campina-grande/), DateRange (Apr 2002, ), isCurrent (true) 

A checagem é feita comparando schoolData.getURL() + schoolData.getRange() com cada um dos elementos da lista school. Se school[i].url != schoolData.getUrl() o score deve ser 0. Se for igual, o score vai depender do match do DateRange.